python/ray/tests/test_get_locations.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import pytest
+import platform
+import time
+
+import ray
+
+
+def test_uninitialized():
+    with pytest.raises(RuntimeError):
+        ray.util.get_object_locations([])
+
+
+def test_get_locations(ray_start_regular):
+    node_id = ray.runtime_context.get_runtime_context().get()["node_id"]
+    sizes = [100, 1000]
+    obj_refs = [ray.put(np.zeros(s, dtype=np.uint8)) for s in sizes]
+    ray.wait(obj_refs)
+    locations = ray.util.get_object_locations(obj_refs)
+    for idx, location in enumerate(locations):
+        assert location.primary_node_id() == node_id
+        # assert location.object_size() == sizes[idx]
+        assert location.node_ids() == [node_id]
+        assert not location.is_spilled()
+        assert location.spilled_url() is None
+        assert location.spilled_node_id() is None
+
+
+@pytest.mark.skipif(
+    platform.system() == "Windows", reason="Failing on Windows.")
+def test_spilled_locations(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(num_cpus=1, object_store_memory=75 * 1024 * 1024)
+    ray.init(cluster.address)
+    cluster.wait_for_nodes()
+
+    node_id = ray.runtime_context.get_runtime_context().get()["node_id"]
+
+    @ray.remote
+    def task():
+        arr = np.random.rand(5 * 1024 * 1024)  # 40 MB
+        refs = []
+        refs.append([ray.put(arr) for _ in range(2)])
+        ray.get(ray.put(arr))
+        ray.get(ray.put(arr))
+        return refs
+
+    [object_refs] = ray.get(task.remote())
+    ray.wait(object_refs)
+    locations = ray.util.get_object_locations(object_refs)
+    for location in locations:
+        assert location.primary_node_id() == node_id
+        # assert location.node_ids() == [node_id]
```

Comment:
while writing the test I found the locaiton.node_ids is not deteministic. specifically, it may or may not contains node_id. Is this expected? @rkooo567

Reply:
hmm I think it should always contain node id? (at least primary node id should be there right?)